models/repo_list.go

```diff
@@ -153,6 +154,10 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, coun
 		}
 
 		cond = cond.Or(builder.And(builder.Like{"lower_name", opts.Keyword}, builder.In("owner_id", ownerIds)))
+
+		if opts.Collaborate {
+			cond = cond.Or(builder.Expr(`id IN (SELECT repo_id FROM "access" WHERE access.user_id = ? AND owner_id != ?)`, opts.Searcher.ID, opts.Searcher.ID))
```

Comment:
This will match all collaborative repos, even ones that do not match the other options (e.g. `opts.Private`, `opts.Keyword`, ...)